packages/sample/src/main.ts

```diff
@@ -252,34 +253,168 @@ program
         "Read the total current through all servos. Specify --continuous to run continuously",
     )
     .action(async (options) => {
-        let hubs = await openConnectedExpansionHubs();
-        let hub = hubs[0];
+        let [hub, close] = await getExpansionHubOrThrow();
         let isContinuous = options.continuous !== undefined;
 
         runOnSigint(() => {
-            hub.close();
+            close();
         });
 
         await servoCurrent(hub, isContinuous);
-        hub.close();
+        close();
     });
 
 program
     .command("servo <channel> <pulseWidth> [frameWidth]")
     .description("Run a servo with pulse width and optional frame width")
     .action(async (channel, pulseWidth, frameWidth) => {
+        let [hub, close] = await getExpansionHubOrThrow();
         let channelValue = Number(channel);
         let pulseWidthValue = Number(pulseWidth);
         let frameWidthValue = frameWidth ? Number(frameWidth) : 4000;
-        let hubs = await openConnectedExpansionHubs();
-        let hub = hubs[0];
-
         runOnSigint(async () => {
             await hub.setServoEnable(channelValue, false);
-            hub.close();
+            close();
         });
 
         await runServo(hub, channelValue, pulseWidthValue, frameWidthValue);
     });
 
 program.parse(process.argv);
+
+/**
+ * Returns the expansion hub referred to by the options provided to the program.
+ * This method also returns a close method. Other hubs may need to be opened, so
+ * prefer calling the returned close method over closing the hub directly.
+ */
+async function getExpansionHubOrThrow(): Promise<[hub: ExpansionHub, close: () => void]> {
+    let options = program.opts();
+    let serialNumber = options.serial;
+    let moduleAddress = options.address ? Number(options.address) : undefined;
+    let parentAddress = options.parent ? Number(options.parent) : undefined;
+    if (moduleAddress === 0) {
+        throw new Error("0 is not a valid module address");
+    } else if (parentAddress === 0) {
+        throw new Error("0 is not a valid parent address");
+    }
```

Comment:
These lines should enforce a range of 1-255

Reply:
They should also enforce that `moduleAddress` cannot be set without `parentAddress`. Right now, it looks like doing so just leaves `moduleAddress` ignored.